### DISCUSSION TITLE
Use cases ?

### DISCUSSION THREAD

**Discussion by ignatandrei** [maca88/AsyncGenerator#162] [Q&A]
Could you please describe use cases?

**Comment by fredericDelaporte** [comment]
See [NHibernate](https://github.com/nhibernate/nhibernate-core) for an example use case. Almost all of its async API is generated by this generator. Its async tests are generated  by it too.

**Comment by maca88** [comment]
This generator can be useful when building a library that needs to expose both sync and async apis. By using it, you can implement only sync apis (e.g. `ReadFile`) and leave the async apis e.g. `ReadFileAsync`) to be generated by the generator.

In addition to `NHibernate` you can also check the following projects that are also using it (the configuration is inside `AsyncGenerator.yml`):
[NHibernate-Caches](https://github.com/nhibernate/NHibernate-Caches)
[StackExchange.Redis.Resilience](https://github.com/maca88/StackExchange.Redis.Resilience)
[Breeze.NHibernate](https://github.com/maca88/Breeze.NHibernate)
